### ISSUE TITLE
[Warrior] Arms APL improvements

### ISSUE THREAD

**Issue by amulder87** [simulationcraft/simc#3097]
full profile: http://pastebin.com/RHyNg9kt
full profile with cooldown_react in use: http://pastebin.com/PLB1eHNE

I made some optimizations for the Arms APL. It's based on adjusting the T19H profile, but should work fine with P and M as well. The first profile listed follows the current profile's standard in that cooldown_react isn't used anywhere. I'd argue it's vital to use it for a bit more realism though and should be required on every MS/CS action. I get the following DPS results simming against the default T19H profile with a 240-360 fight length and 100k iterations:

Current profile: **417,622 DPS**
Adjusted profile excluding food/enchant changes(to highlight rotation changes): **425,143 DPS (+7,521 DPS over previous)**
Adjusted profile: **430,139 DPS (+4,996 DPS over previous)**
Adjusted profile using cooldown_react: **425,037 DPS (-5,102 DPS from previous)**

Food/Enchant changes
- food changed to fishbrul_special
- neck enchant changed to mark_of_the_hidden_satyr

Main actions changes:
- removed the waiting for BL part of BC/Avatar, since BC will wait for a CS to go up anyways, which will take more than a second to do
- BC changed to wait for WB/CS to come off CD if either will be available in under 7s when CS debuff is down
- Avatar changed to try to stay aligned with BC or be used in advance if it'll cover the BC duration
- Potion changed to prefer casting with Avatar + BC + CS, no longer cares about execute range since it has no impact on its damage
- WB+CS changed to refresh on last GCD of CS debuff, not wait til it falls off when they'll do half as much damage

Single actions changes:
- Moved heroic charge(still commented out) to top of APL so it doesn't get delayed when in use
- Changed the logic around CS/MS so CS is prioritized whenever SD is down and it's not the final GCD in a BC, rather than there being a separate action line just for MS to override it as the last GCD in BC
- adjusted FR logic to prioritize it when BC will be off CD soon or when CS debuff is down, removed conditional that allows it to be used when SD is up but CS is already off CD so it never gets used when CS is off CD, and lowered rage threshold from 60 to 50 to decrease rage lost to cap slightly
- added a conditional to MS to prevent its use in the last few seconds before BC becomes available which forces the APL to burn rage before a BC by stacking FR up(to be used inside the BC window) and hitting slam, and saves MS for the opening GCD in a BC to fit 2 MSs in during BL
- Removed WB from rotation, it's better to just use it to prevent CS downtime as last GCD of CS

Execute actions changes:
- Moved heroic charge(still commented out) to top of APL so it doesn't get delayed when in use, also corrected typo where it was prefaced with "actions.single" instead of "actions.execute"
- Lowered execute threshold with PS up to 17.6 instead of 22, the rage needed for a max execute when you have both PS and Dauntless, also allows a stone_heart proc to be used in place of having enough rage here since it's always going to be treated as max rage
- Removed WB from rotation, it's better to just use it to prevent CS downtime as last GCD of CS

**Comment by Collisionc** [comment]
Made a few small changes, but mostly kept it intact. The gcd.remains<0.25 is to push the 22~% haste breakpoint back to 25% or so, which fits in-game rotations better.

**Comment by amulder87** [comment]
Sounds good! I realized afterwards that you'd updated the APL with some changes for archavon's heavy hand and a few other things after the version I was looking at, so feel free to merge them together however makes sense. I spent some time making sure execute ring proc placement fit in well, but I didn't dig into using archavon's heavy hand at all, so anything in this APL with regards to the gloves was just intended to leave existing behavior in place.
